apps/assistants/sync.py

```diff
@@ -190,26 +190,49 @@ def are_files_in_sync_with_openai(assistant: OpenAiAssistant) -> bool:
 
     # ensure files match
     for resource in tool_resources:
-        openai_file_ids = []
-        if resource.tool_type == "code_interpreter":
-            code_interpreter = getattr(assistant_data.tool_resources, "code_interpreter", None)
-            if code_interpreter is not None and code_interpreter.file_ids:
-                openai_file_ids = code_interpreter.file_ids
-            else:
-                openai_file_ids = []
-        elif resource.tool_type == "file_search":
-            openai_vector_store_id = resource.extra.get("vector_store_id")
-            if openai_vector_store_id:
-                vector_store_data = client.beta.vector_stores.files.list(vector_store_id=openai_vector_store_id)
-                openai_file_ids = [file.id for file in getattr(vector_store_data, "data", [])]
-            else:
-                openai_file_ids = []
+        openai_file_ids = _get_tool_file_ids_from_openai(client, assistant_data, resource)
         ocs_file_ids = [file.external_id for file in resource.files.all() if file.external_id]
         if set(ocs_file_ids) != set(openai_file_ids):
             return False
     return True
 
 
+def _get_tool_file_ids_from_openai(client, assistant_data, resource: ToolResources) -> list[str]:
+    """
+    Retrieve file IDs from OpenAI based on the specified tool resource type.
+
+    Args:
+        client: The OpenAI client instance used to interact with the OpenAI API.
+        assistant_data: The assistant data containing tool resources.
+        resource (ToolResources): The tool resource object specifying the type of tool and additional parameters.
+
+    Returns:
+        list[str]: A list of file IDs retrieved from OpenAI.
+
+    The function handles two types of tool resources:
+    - "code_interpreter": Returns file IDs directly from the code interpreter tool resource if available.
+    - "file_search": Retrieves file IDs from the OpenAI vector store using the provided vector store ID.
+    """
+    openai_file_ids = []
+    if resource.tool_type == "code_interpreter":
+        code_interpreter = getattr(assistant_data.tool_resources, "code_interpreter", None)
+        if code_interpreter is not None and code_interpreter.file_ids:
+            return code_interpreter.file_ids
+    elif resource.tool_type == "file_search":
+        openai_vector_store_id = resource.extra.get("vector_store_id")
+        if openai_vector_store_id:
+            kwargs = {}
+            while True:
+                vector_store_data = client.beta.vector_stores.files.list(
+                    vector_store_id=openai_vector_store_id, limit=100, **kwargs
+                )
+                openai_file_ids.extend([file.id for file in getattr(vector_store_data, "data", [])])
+                if not vector_store_data.has_more:
+                    break
+                kwargs["after"] = vector_store_data.last_id
```

Comment:
manual pagination shouldn't be necessary since the openai python library provides auto-paginating iterators: https://github.com/openai/openai-python?tab=readme-ov-file#pagination

I'm not sure why we are doing `getattr(vector_store_data, "data", [])` we should just be able to iterate over the response like we do here: https://github.com/dimagi/open-chat-studio/blob/860fc789a16484b03b2293c844c5a21ac6a742be/apps/assistants/sync.py#L275-L280

Reply:
https://github.com/dimagi/open-chat-studio/pull/846